### PR TITLE
Add failure injection command with enums

### PR DESCRIPTION
**Pull request by julianoes**
In order to test failsafe mechanisms it would be nice to be able to inject failures using a command.

Note that for safetey and security, autopilots should only accept such a failure injection command if this has been specifically enabled using e.g. a parameter or even a flag set at build time.

This commit is just a first suggestion how this could look like and in no means exhaustive.

### CODE REVIEW

**Comment by hamishwillee** [comment]
As discussed in dev call.
- This looks good for me, modulo a way to make sure it cannot be invoked by accident. 
- @auturgy Suggests there are possibly ArduPilot messages that do same thing. He will provide info so we can evaluate/compare.

**Comment by dagar** [comment]
Looks good to me, but can we wait for @auturgy to make sure we don't introduce unnecessary duplication?

**Comment by auturgy** [comment]
Ours are parameters not messages, so no duplicates 

Regards,

James

**Comment by hamishwillee** [comment]
OK, merging as WIP.

@auturgy Can you help us understand what, if anything, in the ArduPilot parameters is not covered by this message? It would be good if this could offer an alternative path for ArduPilot in future.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4285,6 +4293,52 @@
       <entry value="2048" name="AIS_FLAGS_VALID_CALLSIGN"/>
       <entry value="4096" name="AIS_FLAGS_VALID_NAME"/>
     </enum>
+    <enum name="FAILURE_UNIT" bitmask="true">
+      <!-- This enum is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
```
Comment (amilcarlucas) [inline message_definitions/v1.0/common.xml:4297]: This comment is NOT enough. You need to add a <wip> xml tag

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -4285,6 +4293,52 @@
...
+      <entry value="6" name="FAILURE_UNIT_SENSOR_VIO"/>
+      <entry value="7" name="FAILURE_UNIT_SENSOR_DISTANCE_SENSOR"/>
+      <entry value="100" name="FAILURE_UNIT_SYSTEM_BATTERY"/>
+      <entry value="101" name="FAILURE_UNIT_SYSTEM_MOTOR"/>
+      <entry value="102" name="FAILURE_UNIT_SYSTEM_SERVO"/>
+      <entry value="103" name="FAILURE_UNIT_SYSTEM_AVOIDANCE"/>
+      <entry value="104" name="FAILURE_UNIT_SYSTEM_RC_SIGNAL"/>
+      <entry value="105" name="FAILURE_UNIT_SYSTEM_MAVLINK_SIGNAL"/>
+    </enum>
+    <enum name="FAILURE_TYPE" bitmask="true">
```
Comment (amilcarlucas) [inline message_definitions/v1.0/common.xml:4314]: If this is a bitmask, the values need to be a power of two.

File: message_definitions/v1.0/common.xml [hunk 3]
```diff
@@ -4285,6 +4293,52 @@
       <entry value="2048" name="AIS_FLAGS_VALID_CALLSIGN"/>
       <entry value="4096" name="AIS_FLAGS_VALID_NAME"/>
     </enum>
+    <enum name="FAILURE_UNIT" bitmask="true">
```
Comment (amilcarlucas) [inline message_definitions/v1.0/common.xml:4296]: This is not a bitmask!!!

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:4297]: @amilcarlucas You can't add WIP to an enum - only an enum entry. So this is "best that can be done right now".

That seems like a flaw in the XML schema - can you add that support so we can update these properly?

File: message_definitions/v1.0/common.xml (hunk 2 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:4314]: Thanks. Fixed by https://github.com/mavlink/mavlink/pull/1338

File: message_definitions/v1.0/common.xml (hunk 3 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:4296]: Thanks. Fixed by https://github.com/mavlink/mavlink/pull/1338

**Comment by julianoes** [comment]
Thanks for the review @amilcarlucas and thanks for the fix @hamishwillee!

**Pull request merged**